### PR TITLE
[Core] remove compiler warnings intel 858

### LINKED ISSUE #1973: Many compiler warnings with Intel Compiler on Linux
When compiling with the Intel compiler there are a lot of compiler warnings like the following:
~~~
In file included from /home/hpc/pr94va/ga59vax3/software/Kratos/kratos/includes/variables.h(29),
                 from /home/hpc/pr94va/ga59vax3/software/Kratos/kratos/includes/constitutive_law.h(28),
                 from /home/hpc/pr94va/ga59vax3/software/Kratos/applications/StructuralMechanicsApplication/custom_python/add_custom_constitutive_laws_to_python.cpp(20):
/home/hpc/pr94va/ga59vax3/software/Kratos/kratos/containers/vector_component_adaptor.h(127): warning #858: type qualifier on return type is meaningless
      const int GetComponentIndex() const {
      ^
~~~

Does this also appear with other compilers?
Should we remove those?

### PR DESCRIPTION
**Pull request by philbucher**
This PR removes some unnecessary `const` that cause many compiler warnings with the intel compiler ( see #1973 )

closes #1973

### CODE REVIEW

**Comment by philbucher** [comment]
sorry for the whitespace changes

**Comment by philbucher** [comment]
build didn't succeed bcs tests in DEM took too long, restarted it

File: applications/StructuralMechanicsApplication/custom_utilities/shellq4_local_coordinate_system.hpp [hunk 1]
```diff
@@ -308,7 +308,7 @@ namespace Kratos
 		}
 
 		template<int TComponent>
-		inline const double Coord_ij(int Tid1, int Tid2)const {
```
Comment (pooyan-dadvand) [inline applications/StructuralMechanicsApplication/custom_utilities/shellq4_local_coordinate_system.hpp:311]: why removing the inline

**Review by pooyan-dadvand — DISMISSED** [review]
👍

The removed inline is minor but just to be sure you did it intentionally.

File: applications/StructuralMechanicsApplication/custom_utilities/shellq4_local_coordinate_system.hpp (hunk 1 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_utilities/shellq4_local_coordinate_system.hpp:311]: good spotting, this was not intentional!

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Pull request merged**